generator/discovery/openstack/vm_images_discovery.py

```diff
@@ -0,0 +1,755 @@
+                [
+                    const.CONFIG_CLOUD_RESOURCES,
+                    const.CONFIG_OWN_IMAGES,
+                    os_image.name,
+                    const.CONFIG_LICENSE,
+                ]
+            )
+        except KeyError:
+            # license owner not found in config, use default one
+            gx_image.license = gx_image.operatingSystem.license
+
+    def _add_resource_policy(self, os_image: OS_Image, gx_image: GX_Image) -> None:
+        # read mandatory attributes from config or use default values
+        try:
+            # check if comfig contains image's specific resource policy
+            gx_image.resourcePolicy = self.conf.get_value(
+                [
+                    const.CONFIG_CLOUD_RESOURCES,
+                    const.CONFIG_OWN_IMAGES,
+                    os_image.name,
+                    const.CONFIG_RESOURCE_POLICY,
+                ]
+            )
+        except KeyError:
+            # license owner not found in config, use default one
+            gx_image.resourcePolicy = const.DEFAULT_RESOURCE_POLICY
+
+    @staticmethod
+    def _get_license_list(licenses: List[str]) -> List[Union[str, SPDX]]:
+        license_list = list()
+        for lic in licenses:
+            if lic.startswith("http"):
+                license_list.append(URI(lic))
+            else:
+                license_list.append(lic)
+        return license_list
+
+    @staticmethod
+    def _get_video_ram_size(os_image: OS_Image) -> MemorySize:
+        if os_image.hw_video_ram:
+            return MemorySize(value=float(os_image.hw_video_ram), unit=const.UNIT_MB)
+
+    @staticmethod
+    def _get_update_strategy(os_image: OS_Image) -> UpdateStrategy:
+        if os_image.properties and "replace_frequency" in os_image.properties:
+            update_strategy = UpdateStrategy()
+            update_strategy.replaceFrequency = UpdateFrequency(
+                os_image.properties["replace_frequency"]
+            )
+            update_strategy.oldVersionsValidUntil = os_image.properties["uuid_validity"]
+            update_strategy.providedUntil = os_image.properties["provided_until"]
+            if "hotfix_hours" in os_image.properties:
+                update_strategy.hotfixHours = os_image.properties["hotfix_hours"]
+            return update_strategy
+
+    @staticmethod
+    def _get_description(os_image: OS_Image) -> str:
+        if os_image.properties and "image_description" in os_image.properties:
+            if "managed_by_VENDOR" in os_image.properties:
+                return (
+                    os_image.properties["image_description"]
+                    + " Managed by "
+                    + os_image.properties["managed_by_VENDOR"]
+                )
+            else:
+                return os_image.properties["image_description"]
+
+    @staticmethod
+    def _get_name(os_image: OS_Image) -> str:
+        if os_image.name:
+            return os_image.name
```

Comment:
```suggestion
        return os_image.name or None
```
but the `or None` is only necessary if the `os_image.name` can indeed be an empty string!

Reply:
`os_image.name` can be an empty string, indeed!